### DISCUSSION TITLE
Closest Named Color app doesn't work when ran locally

### DISCUSSION THREAD

**Discussion by P-Star7** [color-js/color.js#264] [Q&A]
I was trying to reverse-engineer this great app for my own purposes, so I downloaded the main color.js branch's source code. But even when unmodified, running the index.html inside the "apps/named" folder results in this:
![image](https://user-images.githubusercontent.com/26371167/209455828-085a915b-646c-424e-bdff-e8baa728db07.png)
Am I doing something wrong?

**Comment by LeaVerou** [comment]
It's hard to figure out what's going on from that description. Does the console include any error messages? Are you running it from localhost or `file://`?

**Comment by facelessuser** [comment]
You can't just open the HTML. For best results, you need to run a simple  HTML server to get the site up and running locally.

Try installing a simple HTTP server:

```console
 $ npm install http-server -g
```

And then run the server:

```console
$ http-server
```

You'll be able to navigate the whole site as seen below, though you could configure the port to something else if you want:

```
http://localhost:8080/
```
<img width="1349" alt="Screenshot 2022-12-25 at 12 22 01 PM" src="https://user-images.githubusercontent.com/1055125/209479801-94501bb5-8ea5-4cdc-8f3f-a31bd397b4a4.png">

**Comment by P-Star7** [comment]
> 

Where do I run these commands? I'm doing them in the colorjs main folder and I keep getting a module not found error, even though I have node.js/npm and color.js installed in that folder, I believe. I'm using cmd for Windows 10 x64 - the shortcut that Node.js installs for you - if that helps.

**Comment by facelessuser** [comment]
@P-Star7 I do them right in the main directory of the repository.

`npm install http-server -g` should install the `http-server` module globally. Then it should be accessible anywhere.

I will admit though that I don't normally do this in Windows (I'm on macOS) so I'm not sure if there are some Windows specific considerations that I'm missing. 

I'm also not sure if the authors of color.js use a different way to run and test their site, this is just what I do on the occasions when I contribute to this project.

**Comment by P-Star7** [comment]
Aha! It now works, I forgot to install it globally before running http-server. I installed it globally afterwards, but never tried http-server again, only "node http-server". Just running http-server did the trick. Thanks!

**Comment by P-Star7** [comment]
That's odd... I spoke too soon. It indeed runs on localhost now, but still with the result in the OP. All the src dependencies in this demo's index.js (index.js and keywords.js) are where they SHOULD be.

**Comment by P-Star7** [comment]
Okay, now it does. Augh...

**Comment by LeaVerou** [comment]
I just enabled [Discussions](https://github.com/LeaVerou/color.js/discussions) on this repo, since it's much more suitable for these kinds of things (troubleshooting as opposed to a bug report). I will transfer this to that section now.
